src/main/java/pl/psnc/dei/service/context/ContextService.java

```diff
@@ -0,0 +1,15 @@
+package pl.psnc.dei.service.context;
+
+import pl.psnc.dei.model.Record;
+
+/**
+ * Service used for manipulation on concrete service
+ * @param <T> context class being served by this service
+ */
+public abstract class ContextService<T> {
+    public abstract T get (Record record);
+    public abstract T save (T context);
+    public abstract void delete(T context);
+    public abstract Boolean canHandle(Record record);
```

Comment:
Why not use the primitive boolean?

Reply:
First-class citizen, cache, no boxing-unboxing